### PR TITLE
Ticket: 26659, two scenarios are covered. Spare disks are not replace…

### PR DESCRIPTION
**Pull request by sonicaj**
…able and shouldnot display the replace button. Secondly any disk which is attached to any volume can not be used for replacement

### CODE REVIEW

File: gui/middleware/zfs.py [hunk 1]
```diff
@@ -272,9 +271,7 @@ def get_disks(self):
         """
         disks = []
         for vdev in self:
-            for disk in vdev:
-                if disk.disk:
-                    disks.append(disk.disk)
+            disks.extend(vdev.get_disks())
```
Comment (william-gr) [inline gui/middleware/zfs.py]: Thats not the same, `disk.disk` could be None.

File: gui/middleware/zfs.py [hunk 2]
```diff
@@ -291,18 +288,39 @@ class Vdev(Tnode):
     def __repr__(self):
         return "<Section: %s>" % self.name
 
+    def get_disks(self):
+        disks = []
+        for child in self:
+            if isinstance(child, Vdev):
+                disks.extend(child.get_disks())
+            else:
+                if child.disk:
```
Comment (sonicaj) [inline gui/middleware/zfs.py]: I am checking the disk here if it is None or not?

Comment (sonicaj) [reply gui/middleware/zfs.py]: Would this work?

Comment (william-gr) [reply gui/middleware/zfs.py]: I deleted my comment a few minutes ago ; )

Comment (sonicaj) [reply gui/middleware/zfs.py]: Oh : )

**Comment by william-gr** [comment]
rebuild

**Comment by william-gr** [comment]
rebuild

**Pull request merged**